src/main/java/org/isf/opd/rest/OpdController.java

```diff
@@ -101,6 +120,45 @@ ResponseEntity<OpdDTO> newOpd(@RequestBody OpdDTO opdDTO) throws OHServiceExcept
 		return ResponseEntity.status(HttpStatus.CREATED).body(mapper.map2DTO(isCreatedOpd));
 	}
 
+	/**
+	 * Create a new {@link OpdWithOperatioRowDTO}.
+	 * @param OpdWithOperatioRowDTO
+	 * @return the OpdWithOperatioRowDTO stored
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/opds/rows", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<OpdWithOperatioRowDTO> newOpdWithOperationRow(@RequestBody OpdWithOperatioRowDTO opdWithOperatioRowDTO) throws OHServiceException {
+		int code = opdWithOperatioRowDTO.getOpdDTO().getCode();
+		LOGGER.info("store Out patient {}", code);
+		OpdWithOperatioRowDTO opdWithOperatioRow = new OpdWithOperatioRowDTO();
+		Patient patient = patientManager.getPatientById(opdWithOperatioRowDTO.getOpdDTO().getPatientCode());
+		if (patient == null) {
+			throw new OHAPIException(new OHExceptionMessage(null, "Patient not found!", OHSeverityLevel.ERROR));
```

Comment:
I personally find adding `!` to the end of an error message unnecessary and unprofessional.

Reply:
for that i think it is better to create the task in other to remove all ! to the end of and error message

Reply:
sorry @ArnaudFonzam how many "!" we have in "api" code so far?

Reply:
Ok, the task could be this one https://openhospital.atlassian.net/browse/OH2-182 that will also refactor how the errors are managed in API, without tampering with the core anymore.